tgg/relatedness/n_non_ref_pairs.py

```diff
@@ -0,0 +1,273 @@
+freq_index = get_cohort_index(call_stats_ht)
+var = vds.variant_data.annotate_rows(call_stats=call_stats_ht[vds.variant_data.row_key].freq[freq_index])
+vds = hl.vds.VariantDataset(vds.reference_data, var)
+vds.write(ukb_exomes_path, overwrite=args.overwrite)
+
+Full VDS size: 8.69 TiB
+VDS variant data size: 561.97 GiB
+"""
+
+TRANCHE_DATA = ("broad", CURRENT_FREEZE)
+"""
+UKB tranche data (data source and data freeze number)
+"""
+
+
+def get_n_non_ref_sites(
+    vds_path: str = VDS_PATH,
+    temp_path: str = TEMP_PATH,
+    tranche_data: Tuple[str, int] = TRANCHE_DATA,
+    autosomes_only: bool = True,
+    snp_only: bool = False,
+    het_only: bool = False,
+    adj_only: bool = True,
+    interval_qc_regions: bool = True,
+    no_AS_lowqual: bool = True,
+    non_ref_samples: int = 3,
+) -> hl.Table:
+    """
+    Filter VDS to autosomal sites in interval QC pass regions with an adj non ref of n.
+    :param vds_path: Path to VDS. Default is VDS_PATH.
+    :param temp_path: Path to bucket to store Table and other temporary data. Default is TEMP_PATH.
+    :param tranche_data: UKB tranche data (data source and data freeze number). Default is TRANCHE_DATA.
+    :param autosomes_only: Filter VDS to autosomes. Defaults to True.
+    :param snp_only: Filter VDS to bi-allelic SNPs. Defaults to False.
+    :param het_only: Filter to only het samples. Defaults to False.
+    :param adj_only: Filter GT to adj. Defaults to True.
+    :param interval_qc_regions: Filter to interval QC regions. Defaults to True.
+    :param no_AS_lowqual: Remove AS_lowqual sites. Defaults to True.
+    :param non_ref_samples: Number of non ref samples found in each variant to filter to, e.g. for tripletons, n_samples=3. Defaults to 3.
+    :return: Table of high quality sites with private to n samples.
+    """
+    vds = hl.vds.read_vds(vds_path)
+    mt = vds.variant_data
+
+    if autosomes_only:
+        logger.info("Filter to autosomes...")
+        mt = mt.filter_rows(mt.locus.in_autosome())
+
+    if snp_only:
+        logger.info("Filtering to bi-allelic SNPs...")
+        mt = mt.filter_rows(
+            bi_allelic_expr(mt) & hl.is_snp(mt.alleles[0], mt.alleles[1])
+        )
+
+    if no_AS_lowqual:
+        logger.info("Removing AS_lowqual sites...")
+        info_ht = hl.read_table(info_ht_path(*tranche_data, split=True))
+        mt = mt.filter_rows(~info_ht[mt.row_key].AS_lowqual)
+
+    if interval_qc_regions:
+        logger.info("Filtering to interval QC pass regions...")
+        interval_ht = hl.read_table(interval_qc_path(*tranche_data, "autosomes"))
+        mt = mt.filter_rows(hl.is_defined(interval_ht[mt.locus]))
+
+    if adj_only:
+        logger.info("Filtering to adj and calculating allele count...")
+        mt = filter_to_adj(mt)
+
+    logger.info("Changing old callstat annotation...")
+    mt = mt.transmute_rows(original_call_stats=mt.call_stats)
```

Comment:
this is only required if using the default VDS path, right? maybe this could be
```suggestion
    if "call_stats" in mt.row:
        logger.info("Changing old callstat annotation...")
        mt = mt.transmute_rows(original_call_stats=mt.call_stats)
```